app/models/reservation.rb

```diff
@@ -70,4 +75,18 @@ def move_ended_at_to_end_of_day
   def validate_reservation_dates
     errors.add(:ended_at, "end date must be after the start date") if started_at.present? && ended_at.present? && started_at.to_date >= ended_at.to_date
   end
+
+  def must_have_pickup_event
+    return if manager.pending? || pickup_event.present?
+
+    errors.add(:pickup_event_id, "can't be blank")
+  end
```

Comment:
This seems like a pretty nice way to have validations conditionally apply based on the reservation's state.